runtime/sema/type.go

```diff
@@ -84,6 +84,9 @@ type Type interface {
 	// e.g. in a field of a composite type.
 	IsStorable(results map[*Member]bool) bool
 
+	// IsExternallyReturnable returns true if a value of this type can be exported
+	IsExternallyReturnable(results map[*Member]bool) bool
```

Comment:
What is the input used for? Does it represent members of the type implementing this interface?

Reply:
It's the same as for `exportValueWithInterpreter`: The check if the type is externally returnable is recursive, the "results" parameter prevents cycles: it is checked at the start of the recursively called function, and pre-set before a recursive call.

I'll add comments where this pattern is used to clarify.

Reply:
Added comments in 04c162c757f2a6ea4de1d515d36c092252ec45a7

Reply:
Thanks for adding the comments 🙏